### PR TITLE
BRS-439 Megamenu

### PR DESCRIPTION
**Pull request by DayKirby-OXD**
### Jira Ticket:
BRS-439 Fixed ordering in megamenu 
BRS-392 Added sitemap page

### Jira Ticket URL:
https://bcparksdigital.atlassian.net/browse/BRS-439

### Description:
Refactored megamenu to switch from having different components for desktop and mobile to one responsive component
Fixed ordering of menu items in lower levels
Removed MUI 
Updated design and animations to Harry's specs
Created sitemap version to troubleshoot, so incidentally completed BRS-392
Styled sitemap, linked to it from footer, added mode to mengamenu "responsive" vs "sitemap"

### CODE REVIEW

**Review by cameronpettit — COMMENTED** [review]
Just noticing that the sitewide header above the megamenu with the bcparks logo seems to have disappeared... is this intentional?

**Comment by DayKirby-OXD** [comment]
Looks like maybe something got borked with rebase or something. Will fix and repush

File: src/staging/src/components/megaMenu.js [hunk 1]
```diff
@@ -1,20 +1,267 @@
...
+  const navigatePage = (e, item, menuMode) => {
+    if (menuMode === 'sitemap') {
+      navigate(item.url)
+    } else {
+      menuNavigate(item, menuMode)
+    }
+  }
+
+  const sectionHover = (e, section, menuMode) => {
+    if ((window.screen.width >= 992) && (menuMode !== 'sitemap') && !isTablet) { // otherwise hover triggered in mobile emulator
```
Comment (cameronpettit) [inline src/staging/src/components/megaMenu.js:75]: Something strange here - hover select still happens on screens <922 wide, if mobile emulator is not active.

File: src/staging/src/components/megaMenu.js [hunk 2]
```diff
@@ -1,20 +1,267 @@
...
+      <div className={'menu-toggle menu-mode--' + menuMode + ' menu-toggle--' + (isMenuOpen ? "open" : "closed")}
+        role="button"
+        tabIndex={0}
+        onKeyDown={e => {
+          if ((e.key === 'Enter') || (e.key === ' ')) {
+            toggleMenu(e)
+          }
+        }}
+        onClick={e => toggleMenu(e)}>
+        <nav className="menu-open"><i className="fa fa-bars"></i></nav>
```
Comment (cameronpettit) [inline src/staging/src/components/megaMenu.js:252]: On some screen sizes, this hamburger button seems to be vertically misaligned

File: src/staging/src/styles/megaMenu/megaMenu.scss [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+                display: none;
+            }
+            .menu-close {
+                display:block;
+            }
+        }
+    }
+
+    .mega-menu.menu-mode--responsive.menu-open{
+        height:1000px;
```
Comment (cameronpettit) [inline src/staging/src/styles/megaMenu/megaMenu.scss:94]: There seems to be some unnecessary empty space below the last item of the megamenu on mobile view

**Review by cameronpettit — COMMENTED** [review]
a few more minor nits

File: src/staging/src/styles/megaMenu/megaMenu.scss (hunk 3 above)
Comment (DayKirby-OXD) [reply src/staging/src/styles/megaMenu/megaMenu.scss:94]: yes, added this in discussion with harry. before the menu would end just below the last item and look weird, this way the white bg will fill a phone (but not necessarily over all the content, depending on length). will resolve this, however re-open if you think it needs more discussion

File: src/staging/src/components/megaMenu.js (hunk 1 above)
Comment (DayKirby-OXD) [reply src/staging/src/components/megaMenu.js:75]: that's true, didn't consider that scenario, will disable hover for that case

File: src/staging/src/components/megaMenu.js (hunk 2 above)
Comment (DayKirby-OXD) [reply src/staging/src/components/megaMenu.js:252]: thanks for the catch, will fix

File: src/staging/src/components/megaMenu.js (hunk 1 above)
Comment (DayKirby-OXD) [reply src/staging/src/components/megaMenu.js]: turns out i should have been using window.innerWidth, fixed

File: src/staging/src/components/megaMenu.js (hunk 2 above)
Comment (DayKirby-OXD) [reply src/staging/src/components/megaMenu.js]: this uncovered a lot of issues with the desktop breakpoint. have switched to bootstrap mixins for breakpoints, way better, suggest adopt across site.

File: src/staging/src/styles/megaMenu/megaMenu.scss [hunk 4]
```diff
@@ -0,0 +1,561 @@
...
+
+@media screen and (min-width: $desktopBreakpoint) {
+
+    .navbar--header {
+        max-width: 1140px; 
+        margin: 0px auto;
+        padding: 0px 20px;
+        height: 100px;
+    }
+}
```
Comment (marklise) [inline src/staging/src/styles/megaMenu/megaMenu.scss]: Here and many other places, we should be using bootstrap classes - esp not pixels

**Review by marklise — CHANGES_REQUESTED** [review]

File: src/staging/src/styles/megaMenu/megaMenu.scss (hunk 4 above)
Comment (DayKirby-OXD) [reply src/staging/src/styles/megaMenu/megaMenu.scss]: agreed. suggest we finish this effort in BRS-444

**Review by marklise — APPROVED** [review]
Ticket being spun off.

**Review by cameronpettit — APPROVED** [review]

**Pull request merged**